tests/pom.xml

```diff
@@ -69,7 +67,7 @@ Test cases are run against Docker images of Hono server + (Apache Qpid Dispatch
     -->
     <link.establishment.timeout>5000</link.establishment.timeout>
     <flow.latency>1000</flow.latency>
-    <request.timeout>500</request.timeout>
+    <request.timeout>5000</request.timeout>
```

Comment:
this looks like quite some increase! Any rationale behind that?

Reply:
The increase in the timeout is necessary since Kafka topics are created automatically when they are accessed for the first time. I tried to find the minimum increase for a stable run, but it turned out that CommandAndControlAmqpIT#testSendOneWayCommandSucceeds is not stable yet. I already had a chat with @calohmn but we couldn't figure out yet if there's still something wrong with the test or if the test found a bug.

In addition to that timeout it seems like multiple other timeouts need to be increased as well (I will add them in a separate commit).

But even with the increased timeouts I still get an error on my machine (note that it seems like the test seems to be successful in general since 60/60 commands were received):

`15:38:15.103 [main] INFO  o.e.h.t.amqp.CommandAndControlAmqpIT - commands sent: 60, commands succeeded: 60 after 443 milliseconds
AMQP14:38:15.188 [vert.x-eventloop-thread-0] ERROR io.vertx.core.impl.ContextImpl - Unhandled exception
AMQPjava.lang.IndexOutOfBoundsException: readerIndex: 380, writerIndex: 190 (expected: 0 <= readerIndex <= writerIndex <= capacity(1024))
AMQP	at io.netty.buffer.AbstractByteBuf.checkIndexBounds(AbstractByteBuf.java:112)
AMQP	at io.netty.buffer.AbstractByteBuf.readerIndex(AbstractByteBuf.java:121)
AMQP	at io.vertx.proton.impl.ProtonReadableBufferImpl.position(ProtonReadableBufferImpl.java:156)
AMQP	at org.apache.qpid.proton.engine.impl.FrameWriterBuffer.put(FrameWriterBuffer.java:134)
AMQP	at org.apache.qpid.proton.engine.impl.FrameWriter.writeFrame(FrameWriter.java:114)
AMQP	at org.apache.qpid.proton.engine.impl.TransportImpl.writeFrame(TransportImpl.java:1133)
AMQP	at org.apache.qpid.proton.engine.impl.TransportImpl.processTransportWorkSender(TransportImpl.java:642)
AMQP	at org.apache.qpid.proton.engine.impl.TransportImpl.processTransportWork(TransportImpl.java:541)
AMQP	at org.apache.qpid.proton.engine.impl.TransportImpl.writeInto(TransportImpl.java:369)
AMQP	at org.apache.qpid.proton.engine.impl.TransportOutputAdaptor.pending(TransportOutputAdaptor.java:59)
AMQP	at org.apache.qpid.proton.engine.impl.TransportOutputAdaptor.head(TransportOutputAdaptor.java:80)
AMQP	at org.apache.qpid.proton.engine.impl.SaslImpl$SwitchingSaslTransportWrapper.head(SaslImpl.java:847)
AMQP	at org.apache.qpid.proton.engine.impl.HandshakeSniffingTransportWrapper.head(HandshakeSniffingTransportWrapper.java:151)
AMQP	at org.apache.qpid.proton.engine.impl.TransportImpl.head(TransportImpl.java:1598)
AMQP	at org.apache.qpid.proton.engine.impl.TransportImpl.getOutputBuffer(TransportImpl.java:1541)
AMQP	at io.vertx.proton.impl.ProtonTransport.flush(ProtonTransport.java:269)
AMQP	at io.vertx.proton.impl.ProtonConnectionImpl.flush(ProtonConnectionImpl.java:296)
AMQP	at io.vertx.proton.impl.ProtonDeliveryImpl.flushConnection(ProtonDeliveryImpl.java:137)
AMQP	at io.vertx.proton.impl.ProtonDeliveryImpl.settle(ProtonDeliveryImpl.java:131)
AMQP	at io.vertx.proton.impl.ProtonDeliveryImpl.fireUpdate(ProtonDeliveryImpl.java:162)
AMQP	at io.vertx.proton.impl.ProtonTransport.handleSocketBuffer(ProtonTransport.java:161)
AMQP	at io.vertx.core.net.impl.NetSocketImpl$DataMessageHandler.handle(NetSocketImpl.java:379)
AMQP	at io.vertx.core.net.impl.NetSocketImpl.lambda$new$2(NetSocketImpl.java:101)
AMQP	at io.vertx.core.streams.impl.InboundBuffer.handleEvent(InboundBuffer.java:237)
AMQP	at io.vertx.core.streams.impl.InboundBuffer.write(InboundBuffer.java:127)
AMQP	at io.vertx.core.net.impl.NetSocketImpl.handleMessage(NetSocketImpl.java:357)
AMQP	at io.vertx.core.impl.ContextImpl.executeTask(ContextImpl.java:366)
AMQP	at io.vertx.core.impl.EventLoopContext.execute(EventLoopContext.java:43)
AMQP	at io.vertx.core.impl.ContextImpl.executeFromIO(ContextImpl.java:229)
AMQP	at io.vertx.core.net.impl.VertxHandler.channelRead(VertxHandler.java:163)
AMQP	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
AMQP	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
AMQP	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
AMQP	at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1518)
AMQP	at io.netty.handler.ssl.SslHandler.decodeJdkCompatible(SslHandler.java:1267)
AMQP	at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1314)
AMQP	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:501)
AMQP	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:440)
AMQP	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:276)
AMQP	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
AMQP	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
AMQP	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
AMQP	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
AMQP	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
AMQP	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
AMQP	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
AMQP	at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:792)
AMQP	at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:475)
AMQP	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:378)
AMQP	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
AMQP	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
AMQP	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
AMQP	at java.base/java.lang.Thread.run(Unknown Source)
AMQP14:38:15.212 [vert.x-eventloop-thread-1] INFO  o.e.h.a.c.c.k.KafkaBasedInternalCommandConsumer - no command handler found for command [Command [name: setValue, tenant-id: 12afd436-a6a3-4ffb-8844-a050aebad6ae, device-id: d607cd97-8ea4-47a8-9fa7-254a75bf4f3c, request-id: null]]
AMQP14:38:15.222 [vert.x-eventloop-thread-1] INFO  o.e.h.a.c.c.k.KafkaBasedInternalCommandConsumer - no command handler found for command [Command [name: setValue, tenant-id: 12afd436-a6a3-4ffb-8844-a050aebad6ae, device-id: d607cd97-8ea4-47a8-9fa7-254a75bf4f3c, request-id: null]]
AMQP14:38:15.235 [vert.x-eventloop-thread-1] INFO  o.e.h.a.c.c.k.KafkaBasedInternalCommandConsumer - no command handler found for command [Command [name: setValue, tenant-id: 12afd436-a6a3-4ffb-8844-a050aebad6ae, device-id: d607cd97-8ea4-47a8-9fa7-254a75bf4f3c, request-id: null]]
AMQP14:38:15.253 [vert.x-eventloop-thread-1] INFO  o.e.h.a.c.c.k.KafkaBasedInternalCommandConsumer - no command handler found for command [Command [name: setValue, tenant-id: 12afd436-a6a3-4ffb-8844-a050aebad6ae, device-id: d607cd97-8ea4-47a8-9fa7-254a75bf4f3c, request-id: null]]
15:38:45.106 [main] INFO  o.e.h.tests.IntegrationTestSupport - legacy client's connection to AMQP Messaging Network closed
15:38:45.107 [main] INFO  o.e.h.t.amqp.CommandAndControlAmqpIT - connection to AMQP adapter closed
15:38:45.109 [vert.x-eventloop-thread-2] INFO  o.e.h.tests.IntegrationTestSupport - new client's connection to messaging network closed`

@calohmn also observed errors in the AMQP adapter:

`AMQP21:06:50.122 [vert.x-eventloop-thread-0] DEBUG o.e.h.a.a.VertxBasedAmqpProtocolAdapter - client [container: vert.x-e55e10fd-cb07-45cc-bd53-5d4eeb14b81f] closed connection with error
AMQPio.vertx.core.impl.NoStackTraceThrowable: Error{condition=amqp:decode-error, description='Unknown constructor', info=null}`

or

`AMQP21:06:52.703 [vert.x-eventloop-thread-1] DEBUG o.eclipse.hono.util.HonoProtonHelper - sender link [address: null] is locally open but underlying transport is disconnected
AMQP21:06:52.703 [vert.x-eventloop-thread-1] DEBUG o.e.h.a.a.VertxBasedAmqpProtocolAdapter - waiting for delivery update timed out after 3000ms (link or connection already closed) [Command [name: setValue, tenant-id: 02693623-b8bc-464b-b8c1-bb0a1bdbcf67, device-id: 88fae071-ef79-4dcc-8010-635ffd689310, request-id: 00224]]`

As the investigation is still ongoing I think it makes sense to postpone the merge/review of this PR until we are sure that the test itself is correct.

Hence I will convert it to draft again.

Reply:
ok, makes sense

Reply:
The AMQP errors here have its root in the wrong vert.x context being used when handling the command in the adapter.
I've created PR #2562 for that.